### PR TITLE
Replaces QDateEdit controls with KDateComboBox

### PR DESCRIPTION
**Pull request by renatofilho**
Use KDateComboBox for date pick widgets

### CODE REVIEW

File: client/src/details/details.h [hunk 1]
```diff
@@ -91,6 +91,7 @@ private Q_SLOTS:
 
 private:
     void hideIfUnsupported(QWidget *widget);
+    bool isQtPrivateObject(const QString &objectName) const;
```
Comment (dfaure-kdab) [inline client/src/details/details.h]: Doesn't need to be a member function, just make it file-static in the .cpp

File: client/src/details/opportunitydetails.cpp [hunk 2]
```diff
@@ -48,9 +48,7 @@ OpportunityDetails::OpportunityDetails(QWidget *parent)
     mUi->setupUi(this);
 
     mUi->urllabel->setTextInteractionFlags(Qt::LinksAccessibleByMouse);
-    mUi->next_call_date->calendarWidget()->setVerticalHeaderFormat(QCalendarWidget::ISOWeekNumbers);
```
Comment (dfaure-kdab) [inline client/src/details/opportunitydetails.cpp]: (does the new widget show week numbers too?)

File: client/src/widgets/qdateeditex.cpp [hunk 3]
```diff
@@ -21,61 +21,72 @@
...
 };
 
 /*!
   \reimp
 */
 QDateEditEx::QDateEditEx(QWidget *parent) :
-    QDateEdit(parent), d(new Private(this))
+    KDateComboBox(parent), d(new Private(this))
 {
+    // set object names for all internal children to avoid it to be hidden by Details widget
```
Comment (dfaure-kdab) [inline client/src/widgets/qdateeditex.cpp]: s/it/them/   (children is plural)
s/to be/being/   (avoid is followed by a noun)

File: client/src/widgets/qdateeditex.h [hunk 4]
```diff
@@ -23,9 +23,10 @@
 #ifndef DATEEDITEX_H
 #define DATEEDITEX_H
 
-#include <QDateEdit>
+#include <KDateComboBox>
+
 #include "fatcrmprivate_export.h"
-class FATCRMPRIVATE_EXPORT QDateEditEx : public QDateEdit
+class FATCRMPRIVATE_EXPORT QDateEditEx : public KDateComboBox
```
Comment (dfaure-kdab) [inline client/src/widgets/qdateeditex.h]: Please rename the class then. QDateEditEx is a QDateEdit extension I found online.
If we're now extending KDateComboBox then we need a different widget name.
NullableDateComboBox? The null date feature is what this class adds to KDateComboBox, right?

File: client/src/details/opportunitydetails.cpp (hunk 2 above)
Comment (renatofilho) [reply client/src/details/opportunitydetails.cpp]: yes it shows in a combo box, and allow you to pick a week

File: client/src/widgets/nullabledatecombobox.cpp [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    return QSize(sz.width() + btn->width() + 3, sz.height());
+}
+
+/*!
+  \reimp
+*/
+QSize NullableDateComboBox::minimumSizeHint() const
+{
+    const QSize sz = KDateComboBox::minimumSizeHint();
+    QAbstractButton *btn = lineEdit()->findChild<QAbstractButton*>();
```
Comment (dfaure-kdab) [inline client/src/widgets/nullabledatecombobox.cpp]: Does this ever find something? The clear button isn't a QAbstractButton, is it?

File: client/src/widgets/nullabledatecombobox.cpp [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+#include <QAbstractItemView>
+#include <QAbstractButton>
+#include <QDate>
+#include <QComboBox>
+#include <QMenu>
+
+#include "fatcrm_client_debug.h"
+
+/*!
+  \class DateEditEx dateeditex.h
```
Comment (dfaure-kdab) [inline client/src/widgets/nullabledatecombobox.cpp]: Remove this line, now wrong.

File: client/src/widgets/nullabledatecombobox.cpp [hunk 7]
```diff
@@ -0,0 +1,266 @@
+/**************************************************************************
+**
+** Copyright (c) 2013 Qualiant Software GmbH
+**
+** Author: Andreas Holzammer, KDAB (andreas.holzammer@kdab.com)
+**
+** Contact: Qualiant Software (d.oberkofler@qualiant.at)
```
Comment (dfaure-kdab) [inline client/src/widgets/nullabledatecombobox.cpp]: Remove this line, people shouldn't bother them ;)

File: client/src/widgets/nullabledatecombobox.cpp [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+**
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** If you have questions regarding the use of this file, please contact
+** Qualiant Software at d.oberkofler@qualiant.at.
```
Comment (dfaure-kdab) [inline client/src/widgets/nullabledatecombobox.cpp]: Remove this paragraph

File: client/src/widgets/nullabledatecombobox.cpp (hunk 5 above)
Comment (dfaure-kdab) [reply client/src/widgets/nullabledatecombobox.cpp]: Ah, OK.

**Pull request merged**